### PR TITLE
Update package structure

### PR DESCRIPTION
**Pull request by afauci**
- Rename/reorganize packages
- Consolidate utils

### CODE REVIEW

File: serialization/src/main/java/com/twitter/serial/serializer/BuilderSerializer.java [hunk 1]
```diff
@@ -14,14 +14,13 @@
...
-import com.twitter.serial.serialization.SerializationContext;
-import com.twitter.serial.serialization.base.OptionalFieldException;
-import com.twitter.serial.serialization.base.SerializationException;
-import com.twitter.serial.serialization.base.SerializationUtils;
-import com.twitter.serial.serialization.base.SerializerInput;
+import com.twitter.serial.object.Builder;
+import com.twitter.serial.util.OptionalFieldException;
+import com.twitter.serial.util.SerializationException;
+import com.twitter.serial.util.SerializationUtils;
+import com.twitter.serial.stream.SerializerInput;
```
Comment (cesar1000) [inline serialization/src/main/java/com/twitter/serial/serializer/BuilderSerializer.java]: No big deal, but these are out of order

File: serialization/src/main/java/com/twitter/serial/util/InternalSerialUtils.java [hunk 2]
```diff
@@ -114,13 +184,11 @@ public static int hashCode(@Nullable Object value, @NotNull Object... values) {
         return cast(TrivialComparator.INSTANCE);
     }
 
-    @SuppressWarnings("BlacklistedInterface")
     private static class TrivialComparator<T> implements Comparator<T>, Serializable {
```
Comment (cesar1000) [inline serialization/src/main/java/com/twitter/serial/util/InternalSerialUtils.java]: I don't think we use these comparators?

Comment (afauci) [reply serialization/src/main/java/com/twitter/serial/util/InternalSerialUtils.java]: They're used in CoreSerializers#getComparatorSerializer, though we might be able to remove that if it's not useful

Comment (cesar1000) [reply serialization/src/main/java/com/twitter/serial/util/InternalSerialUtils.java]: It looks like we're only using them in tests? Let's just get rid of them.

**Pull request merged**